src/main/webapp/app/quiz/manage/apollon-diagrams/detail/apollon-diagram-detail.component.ts

```diff
@@ -160,29 +174,37 @@ export class ApollonDiagramDetailComponent implements OnInit, OnDestroy {
      */
     ngOnDestroy() {
         clearInterval(this.autoSaveInterval);
-        if (this.apollonEditor) {
-            this.apollonEditor.destroy();
+        const editor = this.apollonEditor();
+        if (editor) {
+            editor.destroy();
         }
     }
 
     /**
      * Initializes Apollon Editor with UML Model
      * @param initialModel
      */
-    initializeApollonEditor(initialModel: UMLModel) {
-        if (this.apollonEditor) {
-            this.apollonEditor.destroy();
+    async initializeApollonEditor(initialModel: UMLModel) {
+        const currentEditor = this.apollonEditor();
+        if (currentEditor) {
+            currentEditor.destroy();
         }
 
-        this.apollonEditor = new ApollonEditor(this.editorContainer().nativeElement, {
+        const newEditor = new ApollonEditor(this.editorContainer().nativeElement, {
             mode: ApollonMode.Exporting,
             model: initialModel,
             type: this.apollonDiagram()!.diagramType,
             locale: this.translateService.currentLang as Locale,
         });
-        this.apollonEditor.subscribeToModelChange((newModel) => {
+
+        await newEditor.nextRender;
+
+        newEditor.subscribeToModelChange((newModel) => {
             this.isSaved = JSON.stringify(newModel) === this.apollonDiagram()?.jsonRepresentation;
+            this.modelChangeCounter.update((v) => v + 1);
```

Comment:
What does `v` stand for? Can we use a more descriptive name? Why not just reuse `modelChangeCounter` or `counter`?